examples/mongocxx/get_values_from_documents.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/client.hpp>
+
+using namespace bsoncxx;
+
+// Document model, showing array with nested documents:
+//
+// {
+//     "_id" : ObjectId(...),
+//     "messagelist" : [
+//       {
+//         "uid" : 413098706,
+//         "status" : 2,
+//         "msg": "..."
+//       },
+//       ...
+//     ]
+// }
+
+// Construct a document in the format of 'messagelist'
+document::value new_message(int64_t uid, int32_t status, std::string msg) {
+    builder::stream::document doc;
+    return doc << "uid" << uid << "status" << status << "msg" << msg << builder::stream::finalize;
+}
+
+// Insert a document into the database
+void insert_test_data(mongocxx::collection& coll) {
```

Comment:
I'm not in agreement about making insert_one const, but I'd like to defer discussion of that.

I do feel more strongly about having a consistent rule across the codebase for reference versus pointer function parameters, though.  From a quick-and-dirty grep of the source I can find 1 file containing function declarations/definitions with non-const reference parameters, versus 38 files containing function declarations/definitions with non-const pointer parameters, and I'd love to maintain conformance with the existing style where possible.  I'd be happy take a look at the Meyers' recommendation and consider it for the future, but I'd say the project currently inherits the "All parameters passed by reference must be labeled const" rule from claiming to follow Google C++ style.